src/Driver/API/SQLite/UserDefinedFunctions.php

```diff
@@ -11,6 +15,25 @@
  */
 final class UserDefinedFunctions
 {
+    private const DEFAULT_FUNCTIONS = [
+        'sqrt' => ['callback' => [SqlitePlatform::class, 'udfSqrt'], 'numArgs' => 1],
+        'mod'  => ['callback' => [SqlitePlatform::class, 'udfMod'], 'numArgs' => 2],
+        'locate'  => ['callback' => [SqlitePlatform::class, 'udfLocate'], 'numArgs' => -1],
+    ];
+
+    /**
+     * @param Closure(string, callable, int): bool                   $callback
+     * @param array<string, array{callback: callable, numArgs: int}> $additionalFunctions
+     */
+    public static function register(Closure $callback, array $additionalFunctions): void
```

Comment:
Why limit `$callback` to `Closure`? If it was `callable`, there wouldn't be a need to convert `callable` to `Closure` in consumers.

Reply:
It's a purely internal API, so I didn't want to make it wider than necessary. But I can change that if you want. I personally prefer closures because of the scoping issues that array/string callables have.

On 4.0.x, I'd switch both calls to first-class callable syntax which effectively yields a `Closure`.

Reply:
Currently, it's not wide enough as it forces consumers to convert their callable to a closure for no reason.

> I personally prefer closures because of the scoping issues that array/string callables have.

Does it apply here or it's a general preference? I don't understand the problem.

Reply:
> Does it apply here or it's a general preference?

A general preference, it does not apply here.

> I don't understand the problem.

https://3v4l.org/QoWpT

Reply:
> [...] it does not apply here.

In this case, let's make it `callable`. Otherwise, it's sort of a violation of the I from the SOLID principles. The API forces the consumer to pass it a specific implementation of the `callable` while it doesn't use any of its features.